sources/packages/backend/apps/queue-consumers/src/processors/schedulers/workflow/assessment-workflow-queue-retry.scheduler.ts

```diff
@@ -0,0 +1,107 @@
+import {} from "../../models/processors.models";
+import {
+  InjectLogger,
+  LoggerService,
+  ProcessSummary,
+} from "@sims/utilities/logger";
+import {
+  getSuccessMessageWithAttentionCheck,
+  logProcessSummaryToJobLogger,
+} from "../../../utilities";
+import { QueueConfiguration } from "@sims/sims-db";
+
+/**
+ * Retry assessments.
+ */
+@Processor(QueueNames.AssessmentWorkflowQueueRetry)
+export class WorkflowQueueRetryScheduler extends BaseScheduler<void> {
+  constructor(
+    @InjectQueue(QueueNames.AssessmentWorkflowQueueRetry)
+    schedulerQueue: Queue<void>,
+    queueService: QueueService,
+    private readonly workflowEnqueuerService: WorkflowEnqueuerService,
+  ) {
+    super(schedulerQueue, queueService);
+  }
+
+  /**
+   * Process all assessments that were not processed in a period of time.
+   * @param job job information.
+   * @returns processing result.
+   */
+  @Process()
+  async enqueueAssessmentRetryOperations(job: Job<void>): Promise<string[]> {
+    const processSummary = new ProcessSummary();
+    try {
+      processSummary.info("Checking assessments to be queued for retry.");
+      // Check for assessments cancellations to be retried.
+      await this.executeEnqueueProcess(
+        processSummary,
+        this.workflowEnqueuerService.enqueueCancelAssessmentRetryWorkflows.bind(
+          this.workflowEnqueuerService,
+        ),
+      );
+      // Check for assessments to be started.
+      await this.executeEnqueueProcess(
+        processSummary,
+        this.workflowEnqueuerService.enqueueStartAssessmentRetryWorkflows.bind(
+          this.workflowEnqueuerService,
+        ),
+      );
+      return getSuccessMessageWithAttentionCheck(
+        "Process finalized with success.",
+        processSummary,
+      );
+    } catch (error: unknown) {
+      const errorMessage = "Unexpected error while executing the job.";
+      processSummary.error(errorMessage, error);
+    } finally {
+      this.logger.logProcessSummary(processSummary);
+      await logProcessSummaryToJobLogger(processSummary, job);
+      await this.cleanSchedulerQueueHistory();
+    }
+  }
+
+  /**
+   * Enqueues the process and creates a new process summary for the queue process.
+   * @param parentProcessSummary parent process summary.
+   * @param enqueueProcess enqueue process function to be called.
+   */
+  private async executeEnqueueProcess(
+    parentProcessSummary: ProcessSummary,
+    enqueueProcess: (
+      summary: ProcessSummary,
+      retryMaxDate: Date,
+    ) => Promise<void>,
+  ): Promise<void> {
+    try {
+      // Process summary to be populated by each enqueueing workflow call.
+      // In case an unexpected error happen the finally block will still be able to
+      // output the partial information captured by the processSummary.
+      const serviceProcessSummary = new ProcessSummary();
+      parentProcessSummary.children(serviceProcessSummary);
+
+      const queueConfig: QueueConfiguration =
```

Comment:
We can follow the way how polling limit is used for process notifications.

![image](https://github.com/bcgov/SIMS/assets/54600590/8bd6d141-44b3-41d4-b865-1b12ae47ac06)